cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/plugin/PluginInstantiator.java

```diff
@@ -441,6 +441,13 @@ private Object convertValue(TypeToken<?> fieldType, String value) throws Excepti
         }
       }
 
+      // discard decimal point for all non floating point data types
+      if (Long.class.equals(rawType) || Short.class.equals(rawType)
+        || Integer.class.equals(rawType) || Byte.class.equals(rawType)) {
+        int dotIndex = value.lastIndexOf(".");
```

Comment:
Should we error if user is giving actual double values? Since plugin configs can have double values right?